src/sentry/dynamic_sampling/tasks.py

```diff
@@ -0,0 +1,82 @@
+from sentry.dynamic_sampling.rules.helpers.prioritise_project import _generate_cache_key
+from sentry.dynamic_sampling.rules.utils import get_redis_client_for_ds
+from sentry.models import Organization, Project
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.relay import schedule_invalidate_project_config
+from sentry.utils import metrics
+
+CHUNK_SIZE = 1000
+MAX_SECONDS = 60
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.prioritise_projects",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+)  # type: ignore
+def prioritise_projects() -> None:
+    metrics.incr("sentry.tasks.dynamic_sampling.prioritise_projects.start", sample_rate=1.0)
+    with metrics.timer("sentry.tasks.dynamic_sampling.prioritise_projects", sample_rate=1.0):
+        for org_id, projects_with_tx_count in fetch_projects_with_total_volumes().items():
+            process_projects_sample_rates(org_id, projects_with_tx_count)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.process_projects_sample_rates",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+)  # type: ignore
+def process_projects_sample_rates(
+    organization_id: int, projects_with_tx_count: Sequence[Sequence[int]]
+) -> None:
+    """
+    Takes a single org id and a list of project ids
+    """
+    organization = Organization.objects.get_from_cache(id=organization_id)
+    # Check if feature flag is enabled:
+    if features.has("organizations:ds-prioritise-by-project-bias", organization):
+        with metrics.timer("sentry.tasks.dynamic_sampling.process_projects_sample_rates.core"):
+            adjust_sample_rates(organization_id, projects_with_tx_count)
+
+
+def adjust_sample_rates(org_id: int, projects_with_tx_count: Sequence[Sequence[int]]) -> None:
+    """
+    This function apply model and adjust sample rate per project in org
+    and store it in DS redis cluster, then we invalidate project config
+    so relay can reread it, and we'll inject it from redis cache.
+    """
+    projects = []
+    for project_id, count_per_root in projects_with_tx_count:
+        project = Project.objects.get_from_cache(id=project_id)
+        sample_rate = quotas.get_blended_sample_rate(project)
+        if sample_rate is None:
+            continue
+        projects.append(
+            DSProject(id=project_id, count_per_root=count_per_root, blended_sample_rate=sample_rate)
+        )
+    model = AdjustedModel(projects=projects)
+    ds_projects = model.adjust_sample_rates()
+
+    redis_client = get_redis_client_for_ds()
+    for ds_project in ds_projects:
+        # TODO: Check that sample rate between 0 < sample_rate < 1.0
+        # hash, key, value
+        redis_client.set(
+            _generate_cache_key(project_id=ds_project.id, org_id=org_id),
+            ds_project.new_sample_rate,  # redis stores is as string
+            60 * 60 * 24,
+        )
```

Comment:
NIT:
As an alternative we could store all org projects rates in one redis dict.
My thinking is that we manage re balancing of project rates per org so 
all projects in an org will have their rate either generated or not (and the
generation will be at the same time for all projects), this would decrease the number of keys
and the size of the keys in redis (since the key will not need to contain the project id)..... 
(A very small NIT, ignore it if it doesn't make sense).

Reply:
I agree with Radu, we could reduce the key size and partition elements with a hash. 

```python
redis_client.hset(
            _generate_cache_key(org_id=org_id),
            ds_project.id,
            ds_project.new_sample_rate,  # redis stores is as string
)
```

The problem is that TTL is at the hash level and not at the key level but supposing we always insert the new items in the cache at approximately the same time, having the TTL at the hash level shouldn't make that much of a difference.

What do you think?

Reply:
>  As an alternative we could store all org projects rates in one redis dict.

organizations have extremely variable size. You would almost get better balancing across redis hosts by sticking with top-level keys rather than trying to store one hash per org. Unclear if it matters in practice.

Redis' stability is somewhat sensitive to doing long-running, expensive operations, because all operations run on a single thread, sequentially, per-host. Keys that hold less data each typically means shorter operations (but maybe more of them in return)

Reply:
then you practically have a key _per organization_. the keys inside of hashes don't count as keys (for the purposes of sharding across nodes in a redis cluster)